src/gfx_layout.cpp

```diff
@@ -191,7 +180,15 @@ Layouter::Layouter(const char *str, int maxw, TextColour colour, FontSize fontsi
 			if (l == nullptr) break;
 			this->push_back(std::move(l));
 		}
-	} while (c != '\0');
+
+		/* Break out if this was the last line. */
+		if (line_length == std::string_view::npos) {
+			break;
+		}
+
+		/* Go to the next line. */
+		str = str.substr(line_length + 1);
```

Comment:
Conceptually the following might be better as it does not involve a copy. Though I reckon the compiler might be smart enough to not create the copy. On the other hand, `substr` will throw an exception when you pass it an invalid position and `remove_suffix` does not. So maybe the `substr` variant is better as that implicitly performs bound checks.
```suggestion
		str.remove_suffix(line_length + 1);
```

Reply:
A quick godbolt later, and it seems `remove_prefix` (prefix btw, not suffix :P) saves a few opcodes; but `string_view` is cheap, so the difference is rather low. And indeed, optimizer fixes any difference between them :) So will change, just because it looks nicer :) I have more places I can use this :)

(I thought "remove_prefix" removed a prefix-string .. didn't expect it to remove just characters :D TIL!)